packages/market-service/src/osmosis/osmosis.ts

```diff
@@ -49,6 +50,30 @@ export class OsmosisMarketService implements MarketService {
     if (!adapters.assetIdToOsmosis(assetId)) return null
 
     try {
+      const assetReference = fromAssetId(assetId).assetReference
+      if (isOsmosisLpAsset(assetReference)) {
+        /* No market exists for Osmosis pool assets, but we can calculate the 'price' of each pool token
+      by dividing the pool TVL by the total number of pool tokens. */
+
+        const id = getPoolIdFromAssetReference(assetReference)
+        if (!id) return null
+
+        const poolData = await getPool(id)
+        const marketData = await getPoolMarketData(id)
+        if (!(poolData && poolData.total_shares && marketData)) return null
+
+        return {
+          price: bnOrZero(marketData.liquidity)
+            .dividedBy(bnOrZero(poolData.total_shares.amount))
+            .toString(),
```

Comment:
Not sure that's much of an issue given the pool prices, but usually a safer practice since web is eventually consuming the stringified BN and scientific notation tends not to do well with constructing new `BigNumber`s from

```suggestion
            .toFixed(),
```

<img width="456" alt="image" src="https://user-images.githubusercontent.com/17035424/217063832-7243519b-cf17-435e-b276-02aaaa53b3cb.png">

Reply:
👍